src/bp-members/classes/class-bp-members-notices-rest-controller.php

```diff
@@ -0,0 +1,1090 @@
+						$retval = true;
+					}
+				}
+			}
+		}
+
+		/**
+		 * Filter the  notices `get_item` permissions check.
+		 *
+		 * @since 15.0.0
+		 *
+		 * @param bool|WP_Error   $retval  Returned value.
+		 * @param WP_REST_Request $request The request sent to the API.
+		 */
+		return apply_filters( 'bp_members_notices_rest_get_item_permissions_check', $retval, $request );
+	}
+
+	/**
+	 * Create a notice.
+	 *
+	 * @since 15.0.0
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_REST_Response|WP_Error
+	 */
+	public function create_item( $request ) {
+		// Setting context.
+		$request->set_param( 'context', 'edit' );
+
+		$notice_id = bp_members_save_notice(
+			array(
+				'title'    => $request->get_param( 'title' ),
+				'content'  => $request->get_param( 'content' ),
+				'target'   => $request->get_param( 'target' ),
+				'priority' => $request->get_param( 'priority' ),
+				'date'     => $request->get_param( 'date' ),
+				'url'      => $request->get_param( 'action_url' ),
+				'text'     => $request->get_param( 'action_text' ),
+			)
+		);
+
+		if ( is_wp_error( $notice_id ) ) {
+			$notice_id->add_data(
+				array(
+					'status' => 500,
+				)
+			);
+
+			return $notice_id;
+		}
+
+		// The notice we just created will be active.
+		$notice        = bp_members_get_notice( $id );
+		$fields_update = $this->update_additional_fields_for_object( $notice, $request );
+
+		if ( is_wp_error( $fields_update ) ) {
+			return $fields_update;
+		}
+
+		$retval   = $this->prepare_item_for_response( $notice, $request );
+		$response = rest_ensure_response( $retval );
+
+		/**
+		 * Fires after a notice is created via the REST API.
+		 *
+		 * @since 15.0.0
+		 *
+		 * @param BP_Members_Notice $notice   Notice object.
+		 * @param WP_REST_Response  $response The response data.
+		 * @param WP_REST_Request   $request  The request sent to the API.
+		 */
+		do_action( 'bp_members_notices_rest_create_item', $notice, $response, $request );
+
+		return $response;
+	}
+
+	/**
+	 * Check if a given request has access to create a notice.
+	 *
+	 * @since 15.0.0
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_Error|bool
+	 */
+	public function create_item_permissions_check( $request ) {
+		$retval = $this->manage_item_permissions_check( $request );
+
+		/**
+		 * Filter the notices `create_item` permissions check.
+		 *
+		 * @since 15.0.0
+		 *
+		 * @param bool|WP_Error   $retval  Returned value.
+		 * @param WP_REST_Request $request The request sent to the API.
+		 */
+		return apply_filters( 'bp_members_notices_rest_create_item_permissions_check', $retval, $request );
+	}
+
+	/**
+	 * Update a notice.
+	 *
+	 * @since 15.0.0
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_REST_Response|WP_Error
+	 */
+	public function update_item( $request ) {
+		// Setting context.
+		$request->set_param( 'context', 'edit' );
```

Comment:
I think this should be passed by the consumer of the API. We have been removing this hardcoded values in other REST API endpoints.

Reply:
I haven't changed how it currently is in https://github.com/buddypress/BP-REST/blob/master/includes/bp-messages/classes/class-bp-rest-sitewide-notices-endpoint.php#L305

Do you remember why you chose to force the context in the first place?

Reply:
I **incorrectly** assumed that any `WP_REST_Server::EDITABLE` request use the `edit` context, which is not always the case. It's up to the API consumer to decide which data to show.